components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -507,4 +525,46 @@ private boolean isVerifyEmailClaimAvailable(Map<String, String> claims) {
         return (claims.containsKey(IdentityRecoveryConstants.VERIFY_EMAIL_CLIAM) &&
                 Boolean.parseBoolean(claims.get(IdentityRecoveryConstants.VERIFY_EMAIL_CLIAM)));
     }
+
+    private void publishPreVerifyEmailClaimEvent(Map<String, Object> eventProperties)
```

Comment:
Can't we have a single method for all these methods?

Reply:
Fixed in 61d7548fc71aca19ecb53cb10928da2bb00294d6